orc8r/cloud/go/obsidian/access/rest_middleware.go

```diff
@@ -34,83 +34,69 @@ import (
 
 func Middleware(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		if c == nil || c.Request() == nil {
-			return handleError(c, http.StatusBadRequest, "Invalid Request")
+		req := c.Request()
```

Comment:
You are removing c == nil check. Did you make sure that Echo's middleware handlers would never get nil context?

Reply:
I checked a bunch of example echo code, especially from the echo repo. I couldn't find anywhere that they perform a nil check. So I think it's a safe assumption that the context is never passed in as nil (I don't understand a good reason why they would)

In any case, if this does end up causing issues in the future, we can revert this change

Reply:
if it's not explicitly documented, you choices are: leave the check and do not introduce a new unknown or risk crash and deal with prod issue if 2+ year old echo version, we are using has a bug or a specific condition when it may leave context nil.
My instinct would be to choose a safer option, but if you'd rather prefer to be sure, nil CTX is not allowed and be notified if it happens is - it's OK as well. Up to you.

Reply:
Yeah everywhere in their curated middlewares they don't perform the nil check, so I think it's safe enough to go with